### PR TITLE
refactor: remove moment from timeMachine/selectors/index.ts

### LINKED ISSUE #2053: Moment Refactor: remove moment from timeMachine/selectors
We need to remove `moment` from in `timemachine/selectors/index.ts`

### PR DESCRIPTION
**Pull request by ChitlangeSahas**
Closes #2053

part of : #1844

This PR removes `moment` from `src/timeMachine/selectors/index.ts`

### CODE REVIEW

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 55 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=2054)

File: src/timeMachine/selectors/index.ts [hunk 1]
```diff
@@ -293,14 +295,27 @@ export const getStartTime = (timeRange: TimeRange) => {
   }
 }
 
+// this function takes in a string that represents a date, and swaps the timezone to local.
+// if the date passed was 2021-09-02 12:00:00 UTC the function will return 2021-09-02 12:00:00 <Local TZ>
+const setTimeToLocal = (date?: string): Date => {
```
Comment (hoorayimhelping) [inline src/timeMachine/selectors/index.ts:300]: i'm confused about the intent of this.

`valueOf` and `getTime` return the number of milliseconds and seconds in UTC. are you trying to return the number of milliseconds in local time?

Comment (ChitlangeSahas) [reply src/timeMachine/selectors/index.ts:300]: So, when we pass a string like `2019-09-02` into moment, it interprets the timezone as Local.
Whereas the Date package interprets it as UTC time zone. 

This makes a slight discrepancy in the `valueOf` return value. So to fix it, I change the timezone of the passed string to local. Just like the `[setTimeToUTC](https://github.com/influxdata/ui/blob/e198ecfb1341dbbeb2b8a2d9082d718919704810/src/dashboards/selectors/index.ts#L66)`

I found that out after reading this: https://stackoverflow.com/a/31402874

Comment (hoorayimhelping) [reply src/timeMachine/selectors/index.ts:300]: >I'm trying to get the UTC timestamp in milliseconds but for local time

@ChitlangeSahas is the value we have passed in **always** going to be a string like `YYYY-MM-DD` ?

File: src/timeMachine/selectors/index.ts [hunk 2]
```diff
@@ -293,14 +295,27 @@ export const getStartTime = (timeRange: TimeRange) => {
...
+  const localTime = date ? new Date(date) : new Date()
+  localTime.setMinutes(localTime.getMinutes() + getTimezoneOffset())
+
+  return localTime
+}
+
```
Comment (hoorayimhelping) [inline src/timeMachine/selectors/index.ts:306]: I hate to do this so late in the game, but I think this method should be moved into `dateTimeUtils`

Comment (ChitlangeSahas) [reply src/timeMachine/selectors/index.ts:306]: Oh yes totally, i'll move it!

**Review by hoorayimhelping — APPROVED** [review]

**Pull request merged**